src/engine/QueryPlanner.cpp

```diff
@@ -1792,8 +1794,12 @@ auto QueryPlanner::createJoinWithService(
 
   service->setSiblingTree(sibling._qet);
 
-  SubtreePlan plan = makeSubtreePlan<Join>(
-      service->getExecutionContext(), a._qet, b._qet, jcs[0][0], jcs[0][1]);
+  const auto& qec = service->getExecutionContext();
+
+  SubtreePlan plan =
+      jcs.size() == 1
+          ? makeSubtreePlan<Join>(qec, a._qet, b._qet, jcs[0][0], jcs[0][1])
+          : makeSubtreePlan<MultiColumnJoin>(qec, a._qet, b._qet);
```

Comment:
You should add unit tests for this planning (in the QueryPlannerTest.cpp and QueryPlannerTestHelpers.h you can add a matcher for a service clause, have a look at that code, and let me know if there is trouble)

Reply:
I have implemented a simple test for this in this [commit](https://github.com/ad-freiburg/qlever/pull/1341/commits/2afa9797241f87c6ece53dbacb3fa293f9bfd04b), let me know if that is sufficient or provide feedback for further improvements.